Applications/ClientWebPortal/Source/webapp/src/utils/chainable-models/indexed-model.js

```diff
@@ -0,0 +1,76 @@
+import ArrayModel from './array-model';
+import HashMap from 'hashmap';
+import ArrayStringKeyGenerator from 'utils/array-string-key-generator';
+
+export default class {
+  constructor(indices, columnNames, keyGenerator = new ArrayStringKeyGenerator()) {
+    // validate indices
+    let columnLength = columnNames.length;
+    if (indices == null || indices.length > columnNames) {
+      let errorMessage = 'Indices has to be an array of column numbers';
+      throw new TypeError(errorMessage);
+    } else {
+      for (let i=0; i<indices.length; i++) {
+        if (indices[i] < 0 || indices[i] >= columnNames.length) {
+          let errorMessage = 'Index ' + indices[i] + ' is out of bounds';
+          throw new RangeError(errorMessage);
+        }
+      }
+    }
+
+    this.indices = indices;
```

Comment:
You need to make a copy of indices to ensure integrity.

Reply:
Got it.